hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneListStatusHelper.java

```diff
@@ -215,7 +215,9 @@ public Collection<OzoneFileStatus> listStatusFSO(OmKeyArgs args,
         HeapEntry entry = heapIterator.next();
         OzoneFileStatus status = entry.getStatus(prefixKey,
             scmBlockSize, volumeName, bucketName, replication);
-        map.putIfAbsent(entry.key, status);
+        if (!map.containsKey(entry.key)) {
```

Comment:
AFAIK, `putIfAbsent` does the key check before put it to map.

Reply:
> AFAIK, `putIfAbsent` does the key check before put it to map.

@hemantk-12 Thanks for reviewing the PR, This map will contain deleted keys as well and any deleted key entry will represented in map with null value and since we do not want to override null value as it represents key as deleted, hence the reason of choosing `containsKey` check rather using `putIfAbsent` directly because `putIfAbsent` Java implementation is not as per our expectation in case of null values for a key.  You may want to refer: https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#putIfAbsent-K-V-

Reply:
@devmadhuu You can add a code comment above the change, just like you did in https://github.com/apache/ozone/pull/5244#discussion_r1320367698

Reply:
> @devmadhuu You can add a code comment above the change, just like you did in [#5244 (comment)](https://github.com/apache/ozone/pull/5244#discussion_r1320367698)

Thanks @smengcl , I have added the comment.